ssl/ssl_rsa.c

```diff
@@ -1009,8 +1009,9 @@ int SSL_CTX_use_serverinfo_file(SSL_CTX *ctx, const char *file)
         extension = NULL;
     }
 
-    ret = SSL_CTX_use_serverinfo_ex(ctx, version, serverinfo,
-                                    serverinfo_length);
+    if (version != 0)
```

Comment:
Yeah. I know. The same can be said for every OPENSSL_assert() we put into the code...they should never fail. But they do. And we get CVEs from them. I am opposed to OPENSSL_assert() on principle. Normal assert is fine (perhaps combined with an addition "if" with an error return for production builds).